mono/mini/interp/interp.c

```diff
@@ -4235,16 +4235,13 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, FrameClause
 			MINT_IN_BREAK;
 		}
 		MINT_IN_CASE(MINT_LDOBJ_VT) {
-			void *p;
+			int size;
 			c = (MonoClass*)imethod->data_items[* (guint16 *)(ip + 1)];
 			ip += 2;
-			p = sp [-1].data.p;
-			if (!m_class_is_enumtype (c)) {
```

Comment:
we don't need to handle the enum case anymore?

Reply:
I don't understand what this code was supposed to do. Either way we shouldn't be doing these checks at runtime. We should only have two cases : It either is a primitive, and we use specific opcodes for it, or it is a valuetype and it occupies a size on the vt stack